cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,442 @@
+# -*- coding: utf-8 -*-
+#
+# Tests to validate chargeback costs for daily, weekly, monthly rates.
+#
+# All infra and cloud providers support chargeback reports.
+# But, in order to validate costs for different rates, running the tests on just one provider
+# should suffice.
+import math
+from datetime import date
+
+import fauxfactory
+import pytest
+import re
+
+import cfme.intelligence.chargeback.assignments as cb
+import cfme.intelligence.chargeback.rates as rates
+from cfme import test_requirements
+from cfme.base.credential import Credential
+from cfme.common.vm import VM
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.intelligence.reports.reports import CustomReport
+from cfme.markers.env_markers.provider import ONE
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.tier(2),
+    pytest.mark.parametrize('interval', ['Daily', 'Weekly', 'Monthly'],
+        ids=['daily_rate', 'weekly_rate', 'monthly_rate'], scope='module'),
+    pytest.mark.provider([VMwareProvider], selector=ONE,
+                       scope='module',
+                       required_fields=[(['cap_and_util', 'test_chargeback'], True)]),
+    pytest.mark.usefixtures('has_no_providers_modscope', 'setup_provider_modscope'),
+    test_requirements.chargeback,
+]
+
+# Allowed deviation between the reported value in the Chargeback report and the estimated value.
+DEVIATION = 1
+
+divisor = {
+    'Daily': 24,
+    'Weekly': 24 * 7,
+    'Monthly': 24 * 30
+}
+
+
+@pytest.yield_fixture(scope="module")
+def vm_ownership(enable_candu, provider, appliance):
+    """In these tests, chargeback reports are filtered on VM owner.So,VMs have to be
+    assigned ownership."""
```

Comment:
**REQUIRED**
Here and further please use multiline docstrings where it's required https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings